File: hotels-backend/Hotels/Models/Room.cs
using Hotels.ValidationAttributes;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Hotels.Models
{
    public partial class Room
    {
        [Required]
        public int Id { get; set; }

        [Required]
        [StringLength(20)]
        public string RoomNumber { get; set; }

        [Required]
        [RoomType]
        public string RoomType { get; set; }

        [Required]
        [Range(1, 50)]
        public int VacantBeds { get; set; }

        [Required]
        [Range(0, 9999.99)]
        public decimal Cost { get; set; }

        [Url]
        public string Image { get; set; }

        [Required]
        public int HotelId { get; set; }
        public virtual Hotel Hotel { get; set; }

        public virtual ICollection<Reservation> Reservations { get; set; }
        public virtual ICollection<RoomBlock> RoomBlocks { get; set; }
    }
}

File: hotels-admin-frontend/src/app/room.ts
import { Reservation } from "./reservation";

export class Room {
    id: number;
    roomType: string;
    vacantBeds: number;
    cost: number;
    image: string;
    roomNumber: string;
    hotelId: number;

    constructor() {
        this.id = 0;
        this.hotelId = 0;
        this.roomType = 'Standart';
        this.roomNumber = '';
        this.vacantBeds = 0;
        this.cost = 0;
        this.image = '';
    }
}